app/javascript/components/record-actions/link-incident-to-case/component.jsx

```diff
@@ -0,0 +1,294 @@
+
+  const handleOk = () => {
+    dispatch(linkIncidentToCase({ recordType, incident_ids: incident_ids, case_id: selectedCaseId }));
+    dispatch(fetchLinkIncidentToCaseData({}));
+    dispatch(setCaseIdForIncident(selectedCaseId, selectedCaseDisplayId));
+    dispatch(clearDialog());
+  };
+
+  const handleSubmit = useCallback(data => {
+    setRecordTypeValue('cases');
+    dispatch(fetchLinkIncidentToCaseData(data));
+  }, []);
+
+  const tableOptions = {
+    columns: [
+      {
+        name: "case.id",
+        options: {
+          display: false
+        }
+      },
+      {
+        name: "id",
+        options: {
+          display: false
+        }
+      },
+      {
+        label: i18n.t("potential_match.name"),
+        name: "name"
+      },
+      {
+        label: i18n.t("potential_match.case_id"),
+        name: "case_id_display", options: {
+          customBodyRender: (value, tableMeta) => {
+            const { rowData } = tableMeta;
+            return (
+              <ActionButton
+                id="case.case_id_display-button"
+                text={value}
+                type={ACTION_BUTTON_TYPES.default}
+                variant="text"
+                color="primary"
+                noTranslate
+                onClick={() => 
+                 handleRowClick(rowData[1], rowData[3])
+                }
+              />
+            );
+          }
+        }
+
+      },
+      {
+        label: i18n.t("potential_match.child_age"),
+        name: "age"
+      },
+      {
+        label: i18n.t("potential_match.child_gender"),
+        name: "sex"
+      }
+    ],
+    defaultFilters: fromJS({}),
+    recordType: recordTypeValue,
+    targetRecordType: recordTypeValue,
+    bypassInitialFetch: true,
+
+    options: {
+      selectableRows: "none",
+      onCellClick: false,
+      elevation: 0,
+      pagination: true,
+    },
+
+  };
+
+  const onClose = () => {
+    dispatch(fetchLinkIncidentToCaseData({}))
+    close();
+  };
+
+  // const handleSelectedRecords = (index) => {
+  //   if (index[0].length === 1) {
+  //     setSelectedRecords(index);
+  //     const { id, case_id_display } = fetchIdFromPosition(index);
+  //     setSelectedCaseId(id);
+  //     setSelectedCaseDisplayId(case_id_display)
+  //     setDelayStateUpdate(false);
+  //   } else {
+  //     setSelectedRecords(index);
+  //     setDelayStateUpdate(true);
+  //   }
+  // };
+
+  // const fetchIdFromPosition = (index) => {
+  //   if (dataToJS(caseData) && dataToJS(caseData).length > index[0][0]) {
+  //     const object = dataToJS(caseData)[index[0][0]];
+  //     return { id: object.id, case_id_display: object.case_id_display };
+  //   }
+  //   return null;
+  // };
+
+  const onBack = () => {
+    setShowTablel(true);
+    setCaseInfo(undefined);
+  };
+
+  const handleRowClick =(selectedId, selectedDisplayId)=> {
+      setShowTablel(false);
+      const caseDetail = [];
+      const caseDataInJs = dataToJS(caseData)
```

Comment:
No need to use dataToJS here. Just use map or reduce on immutablejs object to get what you need out of it use the .get method